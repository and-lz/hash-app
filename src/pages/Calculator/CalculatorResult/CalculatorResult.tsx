import { Divider } from 'design-system/Divider/Divider'
import { Spacer } from 'design-system/Spaces/Spaces'
import { Heading2 } from 'design-system/Typography/Headings'
import { HighlightText } from 'design-system/Typography/Paragraph'
import { Bold } from 'design-system/Typography/Styles'
import { formatToCurrency } from 'infrastructure/format/currency'
import React from 'react'
import { getFriendlyDayName } from '../Calculator.helpers'

interface ResultProps {
  days: number[]
  data: any
}

function CalculatorResult({ days, data }: ResultProps) {
  return (
    <>
      <Heading2>Você receberá:</Heading2>
      <Divider />
      <Spacer size="normal" />
      {days.map((day: number) => (
        <HighlightText key={day}>
          <span data-testid="day">{getFriendlyDayName(day)}:</span> <br />
          <Bold data-testid="value">{formatToCurrency(data[day])}</Bold>
        </HighlightText>
      ))}
    </>
  )
}

export default CalculatorResult
