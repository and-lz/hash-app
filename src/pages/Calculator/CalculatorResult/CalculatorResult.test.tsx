import React from 'react'
import { render, RenderResult, screen } from '@testing-library/react'
import CalculatorResult from './CalculatorResult'

describe('<CalculatorResult />', () => {
  let component: RenderResult
  beforeEach(() => {
    component = render(
      <CalculatorResult data={{ 1: 3000, 15: 3050 }} days={[1, 15]} />,
    )
  })
  test('Should render with the correct labels', () => {
    const labels = component.queryAllByTestId('day')
    expect(labels[0]).toHaveTextContent(`AmanhÃ£:`)
    expect(labels[1]).toHaveTextContent(`Em 15 dias:`)
  })
  test('Should render with the correct amount', () => {
    const values = component.queryAllByTestId('value')
    expect(values[0]).toHaveTextContent(`R$ 3.000,00`)
    expect(values[1]).toHaveTextContent(`R$ 3.050,00`)
  })
})
