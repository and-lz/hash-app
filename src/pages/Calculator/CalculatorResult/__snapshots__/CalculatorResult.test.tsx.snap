// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalculatorResult /> Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="sc-hKwCoD gjxRVc"
      >
        Você receberá:
      </h1>
      <hr
        class="sc-bdvvaa gnjgvW"
      />
      <hr
        class="sc-gsDJrp JzKSc"
      />
      <p
        class="sc-jRQAMF jcTbXm"
        data-testid="result-row"
      >
        <span>
          Amanhã
          :
        </span>
         
        <br />
        <em
          class="sc-gKckTs gRhrMD"
        >
          R$ 3.000,00
        </em>
      </p>
      <p
        class="sc-jRQAMF jcTbXm"
        data-testid="result-row"
      >
        <span>
          Em 15 dias
          :
        </span>
         
        <br />
        <em
          class="sc-gKckTs gRhrMD"
        >
          R$ 3.050,00
        </em>
      </p>
      <p
        class="sc-jRQAMF jcTbXm"
        data-testid="result-row"
      >
        <span>
          Em 30 dias
          :
        </span>
         
        <br />
        <em
          class="sc-gKckTs gRhrMD"
        >
          R$ 4.000,00
        </em>
      </p>
      <p
        class="sc-jRQAMF jcTbXm"
        data-testid="result-row"
      >
        <span>
          Em 90 dias
          :
        </span>
         
        <br />
        <em
          class="sc-gKckTs gRhrMD"
        >
          R$ 5.000,00
        </em>
      </p>
    </div>
  </body>,
  "container": <div>
    <h1
      class="sc-hKwCoD gjxRVc"
    >
      Você receberá:
    </h1>
    <hr
      class="sc-bdvvaa gnjgvW"
    />
    <hr
      class="sc-gsDJrp JzKSc"
    />
    <p
      class="sc-jRQAMF jcTbXm"
      data-testid="result-row"
    >
      <span>
        Amanhã
        :
      </span>
       
      <br />
      <em
        class="sc-gKckTs gRhrMD"
      >
        R$ 3.000,00
      </em>
    </p>
    <p
      class="sc-jRQAMF jcTbXm"
      data-testid="result-row"
    >
      <span>
        Em 15 dias
        :
      </span>
       
      <br />
      <em
        class="sc-gKckTs gRhrMD"
      >
        R$ 3.050,00
      </em>
    </p>
    <p
      class="sc-jRQAMF jcTbXm"
      data-testid="result-row"
    >
      <span>
        Em 30 dias
        :
      </span>
       
      <br />
      <em
        class="sc-gKckTs gRhrMD"
      >
        R$ 4.000,00
      </em>
    </p>
    <p
      class="sc-jRQAMF jcTbXm"
      data-testid="result-row"
    >
      <span>
        Em 90 dias
        :
      </span>
       
      <br />
      <em
        class="sc-gKckTs gRhrMD"
      >
        R$ 5.000,00
      </em>
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
